compiler/rustc_typeck/src/check/expr.rs

```diff
@@ -1348,29 +1348,44 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         span: Span,
         remaining_fields: FxHashMap<Ident, (usize, &ty::FieldDef)>,
     ) {
-        let tcx = self.tcx;
         let len = remaining_fields.len();
 
         let mut displayable_field_names =
             remaining_fields.keys().map(|ident| ident.as_str()).collect::<Vec<_>>();
 
         displayable_field_names.sort();
 
-        let truncated_fields_error = if len <= 3 {
-            String::new()
+        let truncated_fields_error: String;
+        let remaining_fields_names = if len <= 3 {
+            truncated_fields_error = String::new();
+
+            match len {
+                1 => format!("`{}`", displayable_field_names[0]),
+                2 => {
+                    format!("`{}` and `{}`", displayable_field_names[0], displayable_field_names[1])
+                }
+                3 => format!(
+                    "`{}`, `{}` and `{}`",
```

Comment:
This would look nicer with an Oxford comma IMO.

Reply:
I'm wondering about the general opinion in the Rust community regarding this. If I search through the source code I pretty much see both versions. Maybe this is a different issue altogether? Perhaps you can create an issue and propose something like a general text style guide.
Just for now I will leave it consistent with the other error message's non-comma "and".

Reply:
Right, my comment was meant to apply to both messages, but I didn't realize one was pre-existing. Getting wider consensus and consistency is probably a good idea.

Reply:
No opinion on this, leaving on the PR author's discretion.